src/_rules/background.js

```diff
@@ -37,13 +37,16 @@ export const backgrounds = [
   ['bg-origin-content', { 'background-origin': 'content-box' }],
   ...makeGlobalStaticRules('bg-origin', 'background-origin'),
 
-  //arbitrary
-  [/^bg-\[(.+)]/, ([, p]) => {
+  // matching arbitrary values
+  [/^bg-\[(.+)]/, ([, p, unit, context]) => {
     if (p.startsWith('url')) {
       // Process url(var(--something)) and extract the var itself -> --something
       const urlAsVar = p.match(/url\(var\(([^)]*)/)?.[1];
       return { 'background-image': urlAsVar ? `var(${urlAsVar})` : p };
+    } else if (p.startsWith('var')) {
+      return { 'background-color': p.startsWith('var') ? p : `var(${p})` };
```

Comment:
This won't work as it will only enter this if clause when p starts with "var". 😄
I have this covered in my branch (with color/variable fixes) though... 😄 So you could just remove this part for now. 🙂